_posts/2024-11-15-dynamic-tables.markdown

```diff
@@ -0,0 +1,86 @@
+
+```sql
+CREATE OR REPLACE DYNAMIC TABLE SCHEMA.COMBINED_SOURCES
+    LAG = '15 minutes' 
+    REFRESH_MODE = INCREMENTAL 
+    INITIALIZE = ON_SCHEDULE 
+    WAREHOUSE = WAREHOUSE 
+AS
+    (
+        SELECT * FROM SCHEMA.SOURCE_A
+        UNION ALL
+        SELECT * FROM SCHEMA.SOURCE_B
+        UNION ALL
+        SELECT * FROM SCHEMA.SOURCE_C
+        UNION ALL
+        SELECT * FROM SCHEMA.SOURCE_D
+        UNION ALL
+        SELECT * FROM SCHEMA.SOURCE_E
+    );
+```
+
+Dynamic Tables just made more sense. They work directly in SQL, can still be version-controlled, and offer easy refresh options. This approach kept things flexible and simple without needing more tools.
+
+# Issues and Concerns
+
+In its current state, visibility into Dynamic Table failures are limited. When an upstream data source changes or is no longer available, identifying the root cause of the failure isn’t always straightforward. We’ve explored a few custom solutions to address this, including setting up a scheduled Snowflake task that queries the Dynamic Table refresh history available in Snowflake’s information schema. Whilst these custom setups are easy to implement, they do add some overhead and aren’t quite as streamlined as we’d like.
+
+Dynamic Tables also come with a few functional limitations. For instance, they don’t support self-merges, window functions, or pivot functions. With self-merges and window functions, the lack of incremental support means that Dynamic Tables perform a full refresh rather than an incremental one, which can increase compute time and somewhat detract from the main benefits of Dynamic Tables. Pivot functions, from what we’ve seen so far, are entirely unsupported, though this does prompt a closer look at whether pivoting is truly required in your workflow.
+
+# Takeaways
+
+Switching to Snowflake Dynamic Tables has been a game-changer.
+
+- Reduced complexity: No need for external orchestration tools; everything is managed directly in SQL.
+- Incremental refresh: Easy to control, helping us keep data up-to-date without overcomplicating things.
+- Simple onboarding: New team members can jump in without needing to learn additional tools.
+- Streamlined workflows: We’re focusing more on building robust data solutions rather than wrestling with infrastructure.
+- Cost-effective: No added fees for extra tooling, and we’re using resources more efficiently.
+
+Dynamic Tables have been a great addition, fitting perfectly into our Snowflake environment and team skill set. These new objects are helping us work more effectively, and we’re excited to see how Dynamic Tables continue to support our data strategy moving forward.
```

Comment:
Just add a call to action - 
Check out our Data & AI capability [here](https://www.mechanicalrock.io/our-expertise/data-and-ai).